### PR TITLE
Bump aws-sdk-go to v1.25.43 to support IRSA for pods EKS

### PR DESCRIPTION
**Pull request by nithu0115**
*Issue #, if available:*
N/A 

*Description of changes:*

Bumping aws-sdk-go to v1.25.43 to support IRSA for pods EKS

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ This plugin requires the following permissions:
...
+This plugin uses the AWS SDK Go, and uses its [default credential provider chain](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html). If you are using the plugin on Amazon EC2 or Amazon ECS or Amazon EKS, the plugin will use your EC2 instance role or [ECS Task role permissions](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html) or [EKS IAM Roles for Service Accounts for pods](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html). The plugin can also retrieve credentials from a (shared credentials file)[https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html], or from the standard `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN` environment variables.
```
Comment (PettitWesley) [inline README.md]: @nithu0115 Not your fault, but can you add a commit to fix this bad markdown: `(shared credentials file)[https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html],`

The `[]` and `()` are backwards.

Comment (nithu0115) [reply README.md]: Sure!

File: go.sum [hunk 2]
```diff
@@ -4,6 +4,8 @@ github.com/aws/amazon-kinesis-firehose-for-fluent-bit v0.0.0-20190723183349-6ca3
 github.com/aws/amazon-kinesis-firehose-for-fluent-bit v0.0.0-20190723183349-6ca31170fc03/go.mod h1:jZjLd+hsaK0oNVotfqbMK9gUsIhXvYEl7Z3tdZTYa54=
 github.com/aws/aws-sdk-go v1.20.6 h1:kmy4Gvdlyez1fV4kw5RYxZzWKVyuHZHgPWeU/YvRsV4=
 github.com/aws/aws-sdk-go v1.20.6/go.mod h1:KmX6BPdI08NWTb3/sm4ZGu5ShLoqVDhKgpiN924inxo=
```
Comment (PettitWesley) [inline go.sum]: Old SDK is still in the go.sum

Fix this by running `go mod tidy`

Comment (nithu0115) [reply go.sum]: For some reason, `go mod tidy` does not fix this issue. It looks like, we might need a new release for amazon-kinesis-firehose-for-fluent-bit where this dependency is coming from!

Comment (PettitWesley) [reply go.sum]: So I finally figured this out. By default, go mod uses the latest git tag from a repo- which is `v1.0.0` for firehose. 

But you can tell it to just use the latest master:

```
github.com/aws/amazon-kinesis-firehose-for-fluent-bit@master
```

I ran that on your branch, and it fixed it.

**Pull request merged**